plugins/modules/pnp_workflow_manager.py

```diff
@@ -1259,6 +1189,151 @@ def verify_diff_deleted(self, config):
+          The method returns an instance of the class with updated attributes:
+          - self.device_response: Gets the device_id and stores it
+
+        Example:
+          passing device details and getting pnp device details response
+        """
+        try:
+            response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function='get_device_list',
+                params={"serial_number": serial_number},
+                op_modifies=True,
+            )
+            if response and len(response) == 1:
+                return response
+
+        except Exception as e:
+            msg = "Device with serial number {0} is not found in the inventory".format(str(serial_number))
+            self.log(msg + str(e), "WARNING")
+            self.module.fail_json(msg=msg)
+
+    def get_device_by_id_pnp(self, device_id):
+        """
+        Get the PNP device details using by device id from the Cisco Catalyst Center.
+
+        Parameters:
+        - self (object): An instance of the class containing the method.
+        - device_id (str): Device id contains id from the device data
+        Returns:
+        The method returns an instance of the class with updated attributes:
+        - self.device_response: Gets the device_id and stores it
+
+        Example:
+        passing device details and getting pnp device details response
+        """
+        try:
+            device_details_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function="get_device_by_id",
+                params={"id": device_id},
+                op_modifies=True,
+            )
+            if device_details_response:
+                return device_details_response
+
+        except Exception as e:
+            msg = "Device with device id {0} is not found in the inventory".format(str(device_id))
+            self.log(msg + str(e), "WARNING")
+            self.module.fail_json(msg=msg)
+
+    def add_pnp_device(self, pnp_params):
+        """
+        Add the PNP device in the Cisco Catalyst Center.
+
+        Parameters:
+        - self (object): An instance of the class containing the method.
+        - pnp_params (dict): Device infomation to add in the inventry
+        Returns:
+        The method returns an instance of the class with updated attributes:
+        - self.device_response: Gets the device_id and stores it
+
+        Example:
+        passing device details and getting pnp device details response
+        """
+        try:
+            device_add_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function="add_device",
+                params=pnp_params,
+                op_modifies=True,
+            )
+            if device_add_response:
+                return device_add_response
+
+        except Exception as e:
+            msg = "Unable to add the PNP device {0}.".format(str(pnp_params))
```

Comment:
msg = "Unable to add the PNP device with parameters: {}. Error: {}".format(pnp_params, str(e))

Reply:
updated the message